ci-operator/step-registry/cucushift/installer/check/aws/gp3-throughput/cucushift-installer-check-aws-gp3-throughput-commands.sh

```diff
@@ -0,0 +1,128 @@
+  exit 1
+fi
+
+# Get cluster infrastructure details
+INFRA_ID=$(jq -r '.infraID' "${SHARED_DIR}/metadata.json")
+CLUSTER_ID="${INFRA_ID}"
+REGION=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.aws.region}')
+
+# Handle C2S/SC2S regions
+if [[ "${CLUSTER_TYPE:-}" =~ ^aws-s?c2s$ ]]; then
+  source_region=$(jq -r ".\"${REGION}\".source_region" "${CLUSTER_PROFILE_DIR}/shift_project_setting.json" 2>/dev/null || echo "${REGION}")
+  if [ -n "${source_region}" ] && [ "${source_region}" != "null" ]; then
+    REGION=$source_region
+  fi
+fi
+
+echo "Cluster ID: ${CLUSTER_ID}"
+echo "Region: ${REGION}"
+
+EXPECTED_THROUGHPUT="${AWS_GP3_THROUGHPUT:-}"
+MIN_THROUGHPUT=125
+MAX_THROUGHPUT=2000
+ret=0
+declare -a FAILURE_SUMMARY=()
+
+function verify_nodes() {
+  local role_label="$1"
+  local role_name="$2"
+
+  local nodes
+  nodes=$(oc get nodes -l "${role_label}" -o jsonpath='{.items[*].metadata.name}')
+  if [ -z "${nodes}" ]; then
+    echo "WARNING: No ${role_name} nodes found, skipping"
+    return
+  fi
+
+  for node in ${nodes}; do
+    local instance_id volume_id volume_json volume_type volume_throughput
+
+    instance_id=$(oc get node "${node}" -o jsonpath='{.spec.providerID}' | sed 's|.*/||')
+    if [ -z "${instance_id}" ]; then
+      echo "WARNING: ${role_name} ${node} has no providerID, skipping"
+      continue
+    fi
+
+    volume_id=$(aws ec2 describe-instances --instance-ids "${instance_id}" --region "${REGION}" \
+      --query "Reservations[0].Instances[0].BlockDeviceMappings[?DeviceName==\`/dev/sda1\`].Ebs.VolumeId" --output text)
+    if [ -z "${volume_id}" ] || [ "${volume_id}" == "None" ]; then
+      echo "WARNING: ${role_name} ${node} has no root volume ID, skipping"
+      continue
+    fi
+
+    volume_json=$(aws ec2 describe-volumes --volume-ids "${volume_id}" --region "${REGION}" --output json)
+    if [ -z "${volume_json}" ] || [ "${volume_json}" == "null" ]; then
+      echo "ERROR: Unable to describe volume ${volume_id}"
+      FAILURE_SUMMARY+=("${node}: unable to describe volume")
+      ret=$((ret+1))
+      continue
+    fi
+
+    volume_type=$(echo "${volume_json}" | jq -r '.Volumes[0].VolumeType')
+    volume_throughput=$(echo "${volume_json}" | jq -r '.Volumes[0].Throughput // "N/A"')
+
+    if [ "${volume_type}" != "gp3" ]; then
+      echo "ERROR: ${node} volume type ${volume_type} != gp3"
+      FAILURE_SUMMARY+=("${node}: volume type ${volume_type}")
+      ret=$((ret+1))
+      continue
+    fi
+
+    if [ "${volume_throughput}" == "N/A" ] || [ "${volume_throughput}" == "null" ]; then
+      echo "ERROR: ${node} gp3 volume missing throughput"
+      FAILURE_SUMMARY+=("${node}: throughput missing")
+      ret=$((ret+1))
+      continue
+    fi
+
+    if [ "${volume_throughput}" -lt "${MIN_THROUGHPUT}" ] || [ "${volume_throughput}" -gt "${MAX_THROUGHPUT}" ]; then
+      echo "ERROR: ${node} throughput ${volume_throughput} outside ${MIN_THROUGHPUT}-${MAX_THROUGHPUT}"
```

Comment:
If we're setting an invalid `volume_throughput `, I guess installer should reject it in the initial validation?